module/VuFind/src/VuFind/Db/Table/Comments.php

```diff
@@ -102,14 +103,14 @@ public function getForResource($id, $source = DEFAULT_SEARCH_BACKEND)
      * Delete a comment if the owner is logged in. Returns true on success.
      *
      * @param int                 $id   ID of row to delete
-     * @param \VuFind\Db\Row\User $user Logged in user object
+     * @param UserEntityInterface $user Logged in user object
      *
      * @return bool
      */
     public function deleteIfOwnedByUser($id, $user)
     {
         // User must be object with ID:
-        if (!is_object($user) || !isset($user->id)) {
+        if (!is_object($user) || !($userId = $user->getId())) {
```

Comment:
Looks like `UserEntityInterface` does not have a return type declaration, and the PHPDoc return value is `int`. Should be `?int` for both I guess?

Reply:
Good point. This felt out of scope for here, so I opened #3599 to address it.